pkg/volume/volume.go

```diff
@@ -65,6 +66,12 @@ type Recycler interface {
 	Recycle() error
 }
 
+// Create adds a new resource in the storage provider and creates a PersistentVolume for the new resource.
+// Calls to Create should block until complete.
+type Creater interface {
+	Create() (*api.PersistentVolume, error)
```

Comment:
The provisioner will map what it gets from PVC onto what it knows to
create.  That might start with "class = gold" mapping to netapp while
"class = bronze" maps to NFS on a RasPi.

On Fri, Oct 2, 2015 at 3:32 PM, Paul Morie <notifications@github.com> wrote:

> In pkg/volume/volume.go
> <https://github.com/kubernetes/kubernetes/pull/13650#discussion_r41074722>
> :
>
> > @@ -65,6 +66,12 @@ type Recycler interface {
> >  	Recycle() error
> >  }
> >
> > +// Create adds a new resource in the storage provider and creates a PersistentVolume for the new resource.
> > +// Calls to Create should block until complete.
> > +type Creater interface {
> > +	Create() (*api.PersistentVolume, error)
>
> @thockin <https://github.com/thockin>
>
> The NetApp NFS provisioner is a totally different beast than the GCE
> provisioner. I do not think we should have a single provisioner
> that knows how to provision every volume type.
>
> I wasn't talking about a single provisioner that knows how to provision
> all types. I was saying, we'll have an EFS provisioner and netapp
> provisioner, do we need to differentiate entire plugins for those since
> they're both NFS on the client side (afaik), or do we need to have a
> mapping of the PV concerns to plugins that allows for us to DNR on volume
> plugins just to differentiate provisioners.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13650/files#r41074722>.
>